app/src/main/java/fr/neamar/kiss/dataprovider/SearchProvider.java

```diff
@@ -58,6 +63,21 @@ public void reload() {
             pojo.url = searchProviderUrls.get("Google");
             pojos.add(pojo);
         }
+
+        Pattern p = Pattern.compile(URL_REGEX);
```

Comment:
Nitpick: Please pre-compile this value on class creation by adding it as a `private static final …` line below the definition of `searchProviderUrls` above.